pmd-core/src/main/java/net/sourceforge/pmd/processor/AbstractPMDProcessor.java

```diff
@@ -108,6 +111,13 @@ public void processFiles(RuleSetFactory ruleSetFactory, List<DataSource> files,
             runAnalysis(new PmdRunnable(dataSource, niceFileName, renderers, ctx, rs, processor));
         }
 
+        // in case we analyzed files within Zip Files/Jars, we need to close them now...
+        for (DataSource dataSource : files) {
```

Comment:
maybe we should just make `DataSource` implement `Autoclosable` with a default empty implementation, so we don't have to deal with details for each plausible implementation here…

Reply:
I've introduced a abstract base class, which holds the default implementation. On PMD 7 with Java8, we could move this to the interface.